pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -336,10 +328,38 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) (Results, err
 	return Results{
 		NewNodeClaims: s.newNodeClaims,
 		ExistingNodes: s.existingNodes,
-		PodErrors:     podErrors,
+		PodErrors: lo.MapEntries(podErrors, func(id types.UID, err error) (*corev1.Pod, error) {
+			return podUIDToPod[id], err
+		}),
 	}, ctx.Err()
 }
 
+func (s *Scheduler) trySchedule(ctx context.Context, p *corev1.Pod, podErrors map[types.UID]error) {
```

Comment:
nit: I don't think there's any reason we need to pass in the full set of pod errors right? Can we update the interface to just return an error, and delete the key from pod errors if we return nil up the stack?